lib/spec/openapi/utils.js

```diff
@@ -99,8 +99,24 @@ function plainJsonObjectToOpenapi3 (container, jsonSchema, externalSchemas) {
         const result = {
           in: container,
           name: propertyName,
-          required: jsonSchemaElement.required,
-          schema: jsonSchemaElement
+          required: jsonSchemaElement.required
+        }
+        // complex serialization in query or cookie, eg. JSON
+        // https://swagger.io/docs/specification/describing-parameters/#schema-vs-content
+        if (jsonSchemaElement.consume) {
+          result.content = {
+            [jsonSchemaElement.consume]: {
+              schema: {
+                ...jsonSchemaElement,
+                required: jsonSchemaElement.rawRequired
```

Comment:
`rawRequired` is a interna utility, we could use `Symbols` to hide it from the user

Reply:
ready